pkg/utils/utils_test.go

```diff
@@ -0,0 +1,78 @@
+package utils
```

Comment:
nit: I would name this file nodes_test.go,  so you can have separate test files when you add new util files later on.

Reply:
But nodes_test.go wouldn't be interfered as the test file for nodes.go, while this file shares a test for the taint file as well? ATM it has tests for two utils files, `taint.go`, and `nodes.go`
Going with your direction maybe it would make more sense to move `Context("Taint functioninality test",...` logic to `taint_test.go`, and have `nodes_test.go` for the other tests that utils.go has. 
WDYT?

Reply:
yes, ideally every xyz.go file should have its own xyz_test.go file. But I won't block on this, for these few methods / tests

Reply:
Sounds good to me :)